### PR TITLE
Review assignment/requesting 

### LINKED ISSUE #2281: [APIView] Assign view to architects
Similarly to PR on Github, it should be possible to asign a view to someone. This would allow:
- To know if there is "view that needs architect now" (in combination with #2278)
- To know if Anna/Johan has took ownership of a review

In particular, in the automatic view, the "uploader" column brings no value since it's always azure-sdk. Having an assignee information that we can order would be helpful.

### PR DESCRIPTION
**Pull request by Baas-hub**
Resolves #2281

This adds a dropdown as follows to each review created:
image

You can either check the boxes of each user you want to select, or hit request all and request every listed approver. This will be improved to just be language-specific approvers after #4164. Email notifications to approvers should also come once this issue is resolved.

One a request is submitted, the reviewers get a page as follows:
image

The top table in unapproved reviews, sorted by newest first, and the bottom is approved reviews, sorted by most recently approved. The bottom table is limited to 1 week since approval was given.  

Additional note, once the user profile system is there, the IConfiguration could also be removed once again from Reviews.cshtml.cs and replaced with a different method to get approvers from the profiles.

### CODE REVIEW

File: src/dotnet/APIView/APIViewWeb/Pages/Assemblies/Review.cshtml
Comment (JonathanGiles) [inline src/dotnet/APIView/APIViewWeb/Pages/Assemblies/Review.cshtml:68]: I nice-to-have feature would be to have some JS code that updates a single button (instead of these two) based on whether any users are selected in the dropdown, so that it reads "Request from selected", and if no one is selected, it will read "request from all".

File: src/dotnet/APIView/APIViewWeb/Pages/Assemblies/Review.cshtml
Comment (JonathanGiles) [inline src/dotnet/APIView/APIViewWeb/Pages/Assemblies/Review.cshtml:64]: Even though it is unlikely, we should consider the possibility that there are a lot of reviewers for a language. In this case, having a UI similar to GitHub where the list is scrollable would make sense.

File: src/dotnet/APIView/APIViewWeb/Pages/Assemblies/RequestedReviews.cshtml
Comment (JonathanGiles) [inline src/dotnet/APIView/APIViewWeb/Pages/Assemblies/RequestedReviews.cshtml:24]: For the two tables on this page, let's introduce headers above each of them that read "Pending Reviews" and "Recently-Approved Reviews"

File: src/dotnet/APIView/APIViewWeb/Pages/Assemblies/RequestedReviews.cshtml
Comment (JonathanGiles) [inline src/dotnet/APIView/APIViewWeb/Pages/Assemblies/RequestedReviews.cshtml:124]: Random thought - do we change the links where we link to a user throughout APIView to instead be a link to their user profile page (if one exists, and assuming we build a read-only version of their profile page). This might be useful in the sense that it shows me their email address, as well as a link to their GitHub profile. Something for @chidozieononiwu to consider.

File: src/dotnet/APIView/APIViewWeb/Pages/Assemblies/RequestedReviews.cshtml
Comment (JonathanGiles) [inline src/dotnet/APIView/APIViewWeb/Pages/Assemblies/RequestedReviews.cshtml:134]:
```suggestion
                                <td colspan="6">No reviews have been recently approved.</td>
```

File: src/dotnet/APIView/APIViewWeb/Pages/Assemblies/Review.cshtml
Comment (Baas-hub) [inline src/dotnet/APIView/APIViewWeb/Pages/Assemblies/Review.cshtml:116]: This indentation appears strange, but in Visual Studio is automatically aligned to this

Comment (JonathanGiles) [reply src/dotnet/APIView/APIViewWeb/Pages/Assemblies/Review.cshtml:116]: I'm not behind my PC right now, but is this a spaces vs tabs issue?

Comment (Baas-hub) [reply src/dotnet/APIView/APIViewWeb/Pages/Assemblies/Review.cshtml:116]: Potentially, but wouldn't it occur in more places if it was? I can investigate this possibility soon

File: src/dotnet/APIView/APIViewWeb/Pages/Assemblies/RequestedReviews.cshtml
Comment (chidozieononiwu) [reply src/dotnet/APIView/APIViewWeb/Pages/Assemblies/RequestedReviews.cshtml:126]: Isn't that what happens already? Am I missing something?

File: src/dotnet/APIView/APIViewWeb/Models/ReviewModel.cs
Comment (chidozieononiwu) [inline src/dotnet/APIView/APIViewWeb/Models/ReviewModel.cs:127]: Consider staying with the existing PascalCase style

File: src/dotnet/APIView/APIViewWeb/Pages/Assemblies/RequestedReviews.cshtml
Comment (chidozieononiwu) [inline src/dotnet/APIView/APIViewWeb/Pages/Assemblies/RequestedReviews.cshtml:8]: This is now TempData https://github.com/Azure/azure-sdk-tools/blob/ec2efc121a4fab4d2b03b860dcec98907f6f7d6c/src/dotnet/APIView/APIViewWeb/Pages/Shared/_Layout.cshtml#L43

File: src/dotnet/APIView/APIViewWeb/Pages/Assemblies/Review.cshtml
Comment (chidozieononiwu) [inline src/dotnet/APIView/APIViewWeb/Pages/Assemblies/Review.cshtml:34]: I think we can just joing this with the existing pending/ approved indicator since they don't carry out any other function other than say approved or pending. Perhaps they can also be the dropdown for requesting approval. That can be done as a different set of changes though.

**Review by chidozieononiwu — APPROVED** [review]
You need a rebase, plus I left a few comments. Aside from that I think this is fine.

File: src/dotnet/APIView/APIViewWeb/Pages/Assemblies/RequestedReviews.cshtml
Comment (Baas-hub) [reply src/dotnet/APIView/APIViewWeb/Pages/Assemblies/RequestedReviews.cshtml:126]: When the user profiles pages gets pushed through as we are doing for #4164 currently, it should be doing this

File: src/dotnet/APIView/APIViewWeb/Pages/Assemblies/Review.cshtml
Comment (JonathanGiles) [reply src/dotnet/APIView/APIViewWeb/Pages/Assemblies/Review.cshtml:34]: @chidozieononiwu I'm not fully understanding your suggestion here. If we put the 'Request Review' feature into the 'pending / approved' button, how can we represent both 'pending' state and the ability for people to request review?

Comment (JonathanGiles) [reply src/dotnet/APIView/APIViewWeb/Pages/Assemblies/Review.cshtml:34]: So will it ever say "pending", or will it instead say "request review"? It seems possible that it can be both of those things at the same time. If it says pending, and there is a drop-down arrow to request reviews - do you think that gives enough affordance to users so that they know the functionality is there?

File: src/dotnet/APIView/APIViewWeb/Pages/Assemblies/RequestedReviews.cshtml
Comment (praveenkuttappan) [inline src/dotnet/APIView/APIViewWeb/Pages/Assemblies/RequestedReviews.cshtml:121]: This will fail if approval date is null which is the case of existing approved reviews. right? We need to handle the scenario when review doesn't have approval date (new field)

**Review by praveenkuttappan — CHANGES_REQUESTED** [review]
Please see comment about approval date field for existing reviews?

File: src/dotnet/APIView/APIViewWeb/Pages/Assemblies/RequestedReviews.cshtml
Comment (Baas-hub) [reply src/dotnet/APIView/APIViewWeb/Pages/Assemblies/RequestedReviews.cshtml:121]: This will only happen if a review is requested for an already approved API, but the latest commit also will filter out the cases where the review is requested and no date is listed for approval

File: src/dotnet/APIView/APIViewWeb/Pages/Assemblies/RequestedReviews.cshtml.cs
Comment (praveenkuttappan) [inline src/dotnet/APIView/APIViewWeb/Pages/Assemblies/RequestedReviews.cshtml.cs:30]: I had missed this when I reviewed last time. A better approach is to fetch only required reviews from cosmos DB by adding a new query in cosmos repository class to fetch requested reviews. But this can be done as a separate PR to optimize the code.

**Review by praveenkuttappan — APPROVED** [review]

**Pull request merged**
